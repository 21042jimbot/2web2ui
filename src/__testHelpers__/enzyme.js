export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
