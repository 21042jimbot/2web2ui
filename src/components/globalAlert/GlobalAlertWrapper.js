import React, { Component } from 'react';
import { connect } from 'react-redux';
import TransitionGroup from 'react-transition-group/TransitionGroup';
import { clear } from 'src/actions/globalAlert';
import { Portal, Snackbar } from '@sparkpost/matchbox';

import Animator from './Animator';
import Alert from './Alert';
import styles from './GlobalAlert.module.scss';

export const GlobalAlertWrapper = ({
  alerts,
  clear
}) => (
  <Portal containerId='alert-portal'>
    <div className={styles.Wrapper}>
      <TransitionGroup>
        { alerts.map((alert, i) => (
          <Animator key={`${alert.message}-${alert.date}`}>
            <div className={styles.Alert}>
              <Alert onDismiss={() => clear(i)} {...alert}></Alert>
            </div>
          </Animator>
        ))}
      </TransitionGroup>
    </div>
  </Portal>
);


const mapStateToProps = ({ globalAlert }) => ({ ...globalAlert });
export default connect(mapStateToProps, { clear })(GlobalAlertWrapper);
