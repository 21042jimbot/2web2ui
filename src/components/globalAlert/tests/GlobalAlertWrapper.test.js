import React from 'react';
import { GlobalAlertWrapper } from '../GlobalAlertWrapper';
import { shallow } from 'enzyme';

describe('GlobalAlertWrapper', () => {
  const props = {
    alerts: [{
      message: 'a message',
      id: 'alert_1'
    }],
    clear: jest.fn()
  };

  it('should render', () => {
    const wrapper = shallow(<GlobalAlertWrapper {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should clear', () => {
    const wrapper = shallow(<GlobalAlertWrapper {...props} />);
    wrapper.find('Alert').simulate('dismiss');
    expect(props.clear).toHaveBeenCalledWith('alert_1');
  });
});
