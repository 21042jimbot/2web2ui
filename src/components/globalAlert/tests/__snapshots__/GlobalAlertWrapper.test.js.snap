// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalAlertWrapper should render 1`] = `
<Portal
  containerId="alert-portal"
>
  <div>
    <TransitionGroup
      childFactory={[Function]}
      component="div"
    >
      <Animator
        key="a message-date"
      >
        <div>
          <Alert
            autoDismiss={true}
            date="date"
            message="a message"
            onDismiss={[Function]}
            timeout={15000}
            type="default"
          />
        </div>
      </Animator>
    </TransitionGroup>
  </div>
</Portal>
`;
